app/models/Template.scala

```diff
@@ -0,0 +1,91 @@
+package models
+
+import java.util.UUID
+import play.api.libs.functional.syntax._
+import play.api.libs.json.{JsPath, Json, OWrites, Reads}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+case class MetaData (
+  msName: String,
+  templateName: String,
+  language: Option[List[String]]
+)
+
+object MetaData {
+  
+  implicit val metaDataWrites : OWrites[MetaData] = (
```

Comment:
Since the `JSON` attributes are the same as the `case class` attributes, you could use 
```
import play.api.libs.json._
implicit val metaDataFormat = Json.format[MetaData]
```
That way, you don't have to change all the `Reads` and `Writes`, if the set of attributes changes. You will need explicitly defined implicit `Reads` and `Writes`, if and only if, the attributes set of the `JSON` representation changes.

This holds for all classes here.

Reply:
Cool, i will believe it